tools/check-rec-amendment.js

```diff
@@ -9,8 +9,19 @@ module.exports = async ({github, context, core}) => {
     return;
   }
   const amendments = require(process.env.GITHUB_WORKSPACE + '/amendments.json');
-  if (!Object.values(amendments).find(list => list.find(a => Array.isArray(a.pr) ? a.pr.includes(context.issue.number) : a.pr === context.issue.number ))) {
-     core.setFailed(`Pull request ${context.issue.number} not labeled as editorial and not referenced in amendments.json`);
+  const prAmendmentSection = Object.values(amendments).find(list => list.find(a => Array.isArray(a.pr) ? a.pr.includes(context.issue.number) : a.pr === context.issue.number ));
+  const prAmendment = prAmendmentSection.find(a => Array.isArray(a.pr) ? a.pr.includes(context.issue.number) : a.pr === context.issue.number );
+  if (!prAmendment) {
+    core.setFailed(`Pull request ${context.issue.number} not labeled as editorial and not referenced in amendments.json`);
+  }
+  if (!prAmendment.testUpdates || !prAmendment.testUpdates.length === 0) {
+    core.setFailed(`Pull request ${context.issue.number} declares an amendment but does not document it test status in testUpdates`);
+  }
+  const validTestUpdates = ["already-tested", "not-testable"];
+  if (typeof prAmendment.testUpdates === "string" && !validTestUpdates.includes(prAmendment.testUpdates)) {
+    core.setFailed(`Pull request ${context.issue.number} declares an invalid test status in its amendment testUpdates field`);
```

Comment:
Does it mandate to have the test at the time we add the ammendment?

Reply:
for spec changes that imply a change in the test suite, it requires a link to a WPT pull request at the moment, indeed (but that's distinct from it being implemented anywhere); this could be relaxed, but personally, I think there is real value in requiring an update to the test suite before merging a test-impacting amendment.

Reply:
I understand this but, without an implementation, those tests will be low quality probably, might break and so on.
At least we should flag these tests or make easy to manage/understand they are still experimental.